CONTRIBUTING.md

```diff
@@ -1,30 +1,251 @@
+label – this is appropriate for all contributors. In addition, for those who are relatively new to the open-source
+workflow or our codebase, feel free to view issues tagged with the
+[good first issue](https://github.com/Qiskit-Extensions/quantum-serverless/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
+label.
+
+
+### Setup
+
+So you decided to get your hands dirty and start working on a patch? Then you
+need to know that the project follows the
+[Forking Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow)
+with [Feature Branches](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow).
+
+The above means we expect you to fork the project on your own GitHub account and make your `main` branch to
+track this repository. A typical Git setup after
+[forking the project](https://docs.github.com/en/free-pro-team@latest/github/getting-started-with-github/fork-a-repo) is:
+
+```sh
+# After forking the repository in GitHub
+git clone https://github.com/<your_username>/quantum-serverless.git
+cd quantum-serverless
+git remote add upstream https://github.com/Qiskit-Extensions/quantum-serverless.git
+git remote update upstream
+git checkout main
+git branch -u upstream/main
+git pull
+```
+
+As a core contributor due to some access limitations between forks and the head branch we encourage you to
+[clone](https://support.atlassian.com/bitbucket-cloud/docs/clone-a-repository/) the repository
+instead of forking it.
+
+This repository contains several projects with different technologies. Depending on the project that you selected
+you will run:
+- `pip install -r requirements.txt requirements-dev.txt` for python projects.
+- `helm dependency build` for helm.
+-  `terraform init` for terraform.
+
+
+### Assigning yourself
+
+The very first step to working on an issue is
+[assigning yourself](https://docs.github.com/en/issues/tracking-your-work-with-issues/assigning-issues-and-pull-requests-to-other-github-users#assigning-an-individual-issue-or-pull-request)
+the issue. This gives all contributors the visibility into who is working on what.
+
+### Working on an issue
+
+When you are going to start working on an issue, make sure you are in your `main`
+branch and that it is entirely up-to-date and create a new branch with a
+meaningful name. The typical terminal code for this is:
+
+```sh
+git checkout main
+git pull upstream main
+git checkout -b issue-1234-new-feature
+```
+
+Now start adding your changes and remember to commit often:
+
+```sh
+git commit
+```
+
+And include a summary and some notes or clarifications if needed:
+
+```
+Add a new feature.
+
+The new feature will provide the possibility to do something awesome.
+```
+
+From time to time, you want to check if your `main` branch is still up-to-date. If not, you will need to merge
+(or [rebase](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase)),
```

Comment:
do we want to recommend one over the other? (personally, I prefer rebasing as it keeps the commit history cleaner, but since I _think_ we're doing a squash&merge then it may not ultimately matter...)

Reply:
rebase is my preference as well, what do you think @Tansito @IceKhan13 ?

Reply:
I use them depending of the situation but probably my preference is the rebase too. I can change it.